### PR TITLE
CCD-3600 Ignore NULL fields for Challenge Questions

### PR DESCRIPTION
**Pull request by marvincudjoe**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-3733
https://tools.hmcts.net/jira/browse/CCD-3600

### Change description ###
Ignore NULL fields for Challenge Questions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/ChallengeQuestionParser.java [hunk 1]
```diff
@@ -81,14 +81,25 @@ public ChallengeQuestionTabEntity createChallengeQuestionEntity(ParseContext par
...
+        if (definitionDataItem.getBoolean(ColumnName.CHALLENGE_QUESTION_IGNORE_NULL_FIELDS) == null
+            || !definitionDataItem.getBoolean(ColumnName.CHALLENGE_QUESTION_IGNORE_NULL_FIELDS)) {
+            challengeQuestionTabEntity.setIgnoreNullFields(false);
+            challengeQuestionTabEntity.setAnswerField(definitionDataItem
+                                    .getString(ColumnName.CHALLENGE_QUESTION_ANSWER_FIELD));
+        } else if (Boolean.TRUE.equals(definitionDataItem
+                                    .getBoolean(ColumnName.CHALLENGE_QUESTION_IGNORE_NULL_FIELDS))) {
+            challengeQuestionTabEntity.setIgnoreNullFields(true);
+        }
+    }
```
Comment (danlysiak) [inline excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/ChallengeQuestionParser.java:104]: We don't want any conditional logic for this in the Definition Store - we should always import what is provided (with a default value for `IgnoreNULLFields` set to `false`). The logic for determining whether a nullable answer is acceptable should be considered only in ACA.

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/S-094.1.td.json [hunk 2]
```diff
@@ -36,7 +36,8 @@
 					"display_context_parameter": "#DATETIMEENTRY(dd-MM-yyyy)",
 					"challenge_question_id": "NoCChallenge",
 					"answer_field": "${[scenarioContext][customValues][no_dynamic_injection_${OrganisationField.OrganisationName}|${[scenarioContext][customValues][no_dynamic_injection_${OrganisationField.OrganisationID}:[Claimant],${[scenarioContext][customValues][no_dynamic_injection_${OrganisationField.OrganisationName}|${[scenarioContext][customValues][no_dynamic_injection_${OrganisationField.OrganisationID}:[Defendant]",
-					"question_id": "NoCChallengeQ1"
+					"question_id": "NoCChallengeQ1",
```
Comment (danlysiak) [inline aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/S-094.1.td.json:39]: I think it's worth adding an example where it's `true` here as well - to prove the issue raised in my other comment isn't an issue anymore (after those changes have been made).

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/ChallengeQuestionValidator.java [hunk 3]
```diff
@@ -121,16 +124,30 @@ private void validateQuestionId(DefinitionDataItem definitionDataItem) {
...
+
+    private boolean validateEmptyAnswer(Boolean ignoreNullFields, String answers) {
+        if (answers == null && !ignoreNullFields) {
+            throw new InvalidImportException(ERROR_MESSAGE + " value: answer cannot be null.");
+        }
+        if ((answers == null || answers.isEmpty()) && Boolean.TRUE.equals(ignoreNullFields)) {
+            log.debug("No answer is provided and ignoreNullFields is set to true");
+            return true;
         }
+        return false;
```
Comment (danlysiak) [inline excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/ChallengeQuestionValidator.java:150]: I think we should be able to revert the changes in this class once other comments are addressed.

File: repository/src/main/resources/db/migration/V20220905_3600__CCD-3600_Validation_for_NoC.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE public.challenge_question ADD COLUMN IF NOT EXISTS ignore_null_fields BOOLEAN DEFAULT FALSE;
+ALTER TABLE public.challenge_question ALTER COLUMN answer_field DROP NOT NULL;
```
Comment (danlysiak) [inline repository/src/main/resources/db/migration/V20220905_3600__CCD-3600_Validation_for_NoC.sql:2]:
```suggestion
```

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/ChallengeQuestionParser.java (hunk 1 above)
Comment (padmapriyanalam) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/ChallengeQuestionParser.java]: removed the login in defintion store

File: aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/S-094.1.td.json (hunk 2 above)
Comment (padmapriyanalam) [reply aat/src/aat/resources/features/F-094 NoC ChallengeQuestions/S-094.1.td.json:39]: updated to validate the true condition

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/ChallengeQuestionValidator.java (hunk 3 above)
Comment (padmapriyanalam) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/ChallengeQuestionValidator.java]: reverted the changes
